inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.service.tenant;
+
+import com.github.pagehelper.Page;
+import com.github.pagehelper.PageHelper;
+import com.github.pagehelper.PageInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.inlong.manager.common.consts.InlongConstants;
+import org.apache.inlong.manager.common.enums.ErrorCodeEnum;
+import org.apache.inlong.manager.common.exceptions.BusinessException;
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.dao.entity.InlongTenantEntity;
+import org.apache.inlong.manager.dao.mapper.InlongTenantEntityMapper;
+import org.apache.inlong.manager.pojo.tenant.InlongTenantInfo;
+import org.apache.inlong.manager.pojo.tenant.InlongTenantPageRequest;
+import org.apache.inlong.manager.pojo.tenant.InlongTenantRequest;
+import org.apache.inlong.manager.service.user.LoginUserUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class InlongTenantServiceImpl implements InlongTenantService {
+
+    @Autowired
+    private InlongTenantEntityMapper inlongTenantEntityMapper;
+
+    @Override
+    public InlongTenantInfo get(String name) {
+        InlongTenantEntity entity = inlongTenantEntityMapper.selectByName(name);
+        return CommonBeanUtils.copyProperties(entity, InlongTenantInfo::new);
+    }
+
+    @Override
+    public Integer save(InlongTenantRequest request) {
+        String name = request.getName();
+        InlongTenantEntity existEntity = inlongTenantEntityMapper.selectByName(name);
+        if (existEntity != null) {
+            String errMsg = String.format("tenant already exist for name=%s)", name);
+            log.error(errMsg);
+            throw new BusinessException(errMsg);
+        }
+        InlongTenantEntity entity = CommonBeanUtils.copyProperties(request, InlongTenantEntity::new);
+        String operator = LoginUserUtils.getLoginUser().getName();
+        entity.setCreator(operator);
+        entity.setModifier(operator);
+        inlongTenantEntityMapper.insert(entity);
+        return entity.getId();
+    }
+
+    @Override
+    public PageInfo<InlongTenantInfo> listByCondition(InlongTenantPageRequest request) {
+        PageHelper.startPage(request.getPageNum(), request.getPageSize());
+        Page<InlongTenantEntity> entityPage = inlongTenantEntityMapper.selectByCondition(request);
+        return entityPage
+                .toPageInfo(entity -> CommonBeanUtils.copyProperties(entity, InlongTenantInfo::new));
+    }
+
+    @Override
+    public Boolean update(InlongTenantRequest request) {
+        InlongTenantEntity exist = inlongTenantEntityMapper.selectByName(request.getName());
```

Comment:
Whether the tenant name can be changed？

Reply:
If somebody change the tenant name, all the related resources needs to be changed also. Hence I don't think it's a good idea to allow tenant owner to change tenant name.